smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/fault/KafkaDeadLetterQueue.java

```diff
@@ -141,8 +138,12 @@ private String getThrowableMessage(Throwable throwable) {
                 .thenCompose(m -> record.ack());
     }
 
+    void addHeader(ProducerRecord<?, ?> record, String key, String value) {
+        record.headers().add(key, value.getBytes(StandardCharsets.UTF_8));
+    }
+
     @Override
     public void terminate() {
-        producer.closeAndAwait();
+        producer.close();
```

Comment:
What's the return type of close?

Reply:
void. It'll wait for the duration of `close-timeout` config of the outgoing channel.